options.go

```diff
@@ -37,6 +37,9 @@ type Options struct {
 	// Network and Addr options.
 	Dialer func(ctx context.Context, network, addr string) (net.Conn, error)
 
+	// For sentinel proxy
+	DumpDialer func(ctx context.Context, network, addr string) (net.Conn, error)
```

Comment:
This is wrong place for sentinel specific options. Probably you are looking for FailoverOptions.

Also I don't understand the logic behind `DumpDialer`